### ISSUE TITLE
dnsmasq doing local lookups on external nameservers configured in resolv.conf

### ISSUE THREAD

**Issue by jarkomes** [apprenda/kismatic#489]
dnsmasq tends to search for local domains on the nameservers configured in resolv.conf on the host (in this case google dns) Due to this incorrect lookup, the healthcheck fails and kills the dnsmasq.
```
I0412 08:35:40.180731       1 nanny.go:108] dnsmasq[17]: query[A] kubernetes.default.svc.cluster.local from 136.243.XXX.XXX
I0412 08:35:40.180848       1 nanny.go:108] dnsmasq[17]: forwarded kubernetes.default.svc.cluster.local to 8.8.4.4
I0412 08:35:40.193018       1 nanny.go:108] dnsmasq[17]: reply kubernetes.default.svc.cluster.local is NXDOMAIN
I0412 08:35:40.352488       1 nanny.go:108] dnsmasq[17]: query[A] kubernetes.default.svc.cluster.local from 127.0.0.1
I0412 08:35:40.352517       1 nanny.go:108] dnsmasq[17]: cached kubernetes.default.svc.cluster.local is NXDOMAIN
```
This can simply be resolved by adding the following flag to the dnsmasq-nanny: --no-resolv

(This might be an issue in dnsmasq, but its better to have the flag set...)

**Comment by alexbrand** [comment]
This seems like a regression, as we were originally setting the `--no-resolv` flag.

**Comment by dkoshkin** [comment]
Im not sure about this one, its already being set in kube-dns via https://github.com/kubernetes/dns/pull/59.

Also according to [this](https://github.com/kubernetes/kubernetes/blob/release-1.6/cluster/addons/dns/kubedns-controller.yaml.base) `--no-resolv` is not being set.

**Comment by alexbrand** [comment]
Thinking about this more, I think the upstream DNS server should be reachable, so it seems like we should not set the `--no-resolv` flag.

@jarkomes It seems like something wrong was going on with your kube-dns as it returned `NXDOMAIN` to the `kubernetes.default.svc.cluster.local` query.

**Comment by pcallewaert** [comment]
@alexbrand it was 8.8.8.8 that responded with `NXDOMAIN`, the problem was that a cluster.local url got forwarded to an external DNS. 
But this seems fixed now in v1.3.1 by this commit: https://github.com/apprenda/kismatic/commit/f2d17fc42073f5d3e574aae8122c92d5369d67c3

This forces all cluster.local requests to go to kube-dns, and all other queries will go the the nameservers configured in /etc/resolv.conf, should also work perfectly!

**Comment by alexbrand** [comment]
@pcallewaert Thanks! 

Closing this issue as it should be fixed.
